### PR TITLE
[stable/redis] Fix use password as file environment variable issue

### LINKED ISSUE #10733: [stable/redis] prometheus metrics exporter is crashlooping (missing `/bin/sh`) version 5.3.0
**Version of Helm and Kubernetes**: 

Helm
```
Client: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.6.1", GitCommit:"bbc1f71dc03afc5f00c6ac84b9308f8ecb4f39ac", GitTreeState:"clean"}
```

Kubernetes
```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.0", GitCommit:"0ed33881dc4355495f623c6f22e7dd0b7632b7c0", GitTreeState:"clean", BuildDate:"2018-09-27T17:05:32Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.9-gke.7", GitCommit:"572a070abf31a08a8f0d104220da467ea1e3d5f2", GitTreeState:"clean", BuildDate:"2018-11-14T05:26:58Z", GoVersion:"go1.9.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**: stable/redis

**What happened**: The prometheus metrics exporter pod can't start properly. It's stuck in a crashloop because it's missing `/bin/sh`. We get the following error message:
```
Error: failed to start container "metrics": Error response from daemon: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory"
```

**What you expected to happen**: The pod should start properly.

**How to reproduce it** (as minimally and precisely as possible):
```yaml
- stable/redis:
    version: ~> 5.3.0
    values:
      metrics:
        enabled: true
      usePassword: false
      cluster:
        enabled: false
        slaveCount: 0
```

**Anything else we need to know**:

This bug is introduced in PR https://github.com/helm/charts/pull/10216, the start command for the metrics pod is adjusted but the imaged used `oliver006/redis_exporter` is based on the scratch image so there is no `/bin/sh` available see https://github.com/oliver006/redis_exporter/commit/8c89e535f9aedcd0ca5a170405e287fca643721d

### LINKED ISSUE #10769: [stable/redis] usePasswordFile causes health check failure
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**: Helm 2.12.1, Kubernetes 1.12.4

**Which chart**: stable/redis

**What happened**: I turned on the [`usePasswordFile` value](https://github.com/helm/charts/blob/master/stable/redis/values.yaml#L83), and Redis fails to start up due to health check failures:

> ```Liveness probe failed: ERR unknown command `localhost`, with args beginning with: `-p`, `6379`, `ping`, Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe.```

**What you expected to happen**: Redis starts up and uses the password file

**How to reproduce it** (as minimally and precisely as possible):

Run `helm install stable/redis --values values.yml`

values.yaml:

```yaml
image:
  registry: docker.io
  repository: bitnami/redis
  tag: 5.0.2
  pullPolicy: IfNotPresent
usePasswordFile: true
```

**Anything else we need to know**:

The health checks are relying on password environment variables that don't get set in password-file mode ([usage](https://github.com/helm/charts/blob/master/stable/redis/templates/health-configmap.yaml#L15), [initialization](https://github.com/helm/charts/blob/master/stable/redis/templates/redis-master-statefulset.yaml#L100))

### PR DESCRIPTION
**Pull request by javsalgar**
#### What this PR does / why we need it:
The usePasswordFile feature had issues with environment variables, as they were not being properly evaluated.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
- Fixes https://github.com/helm/charts/issues/10769
- Fixes https://github.com/helm/charts/issues/10733

#### Special notes for your reviewer:
I tested this with both the `/run.sh` command and the `redis-server` command

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by lachie83** [comment]
cc @carrodher @desaintmartin @juan131 @prydonius @sameersbn @tompizmor for review

File: stable/redis/README.md [hunk 1]
```diff
@@ -138,7 +138,7 @@ The following table lists the configurable parameters of the Redis chart and the
 | `master.podLabels`                         | Additional labels for Redis master pod                                                                         | {}                                                   |
 | `master.podAnnotations`                    | Additional annotations for Redis master pod                                                                    | {}                                                   |
 | `master.port`                              | Redis master port                                                                                              | `6379`                                               |
-| `master.command`                           | Redis master entrypoint array. The docker image's ENTRYPOINT is used if this is not provided.                  | []                                                   |
+| `master.command`                           | Redis master entrypoint string. The command `redis-server` is executed if this is not provided.                  | []                                                   |
```
Comment (desaintmartin) [inline stable/redis/README.md]: Default value should be "/run.sh".

**Comment by javsalgar** [comment]
This PR https://github.com/oliver006/redis_exporter/pull/231 will also add the REDIS_PASSWORD_FILE support to the metrics

File: stable/redis/README.md (hunk 1 above)
Comment (javsalgar) [reply stable/redis/README.md]: Fixed. Thanks

**Review by tompizmor — APPROVED** [review]
/ok-to-test
/lgtm

**Comment by tompizmor** [comment]
Waiting until new version of redis-exporter is released

**Comment by javsalgar** [comment]
Let's wait until we have the new redis exporter container released

**Comment by javsalgar** [comment]
@tompizmor the new redis_exporter container was released, we can go ahead with the merging process

**Comment by desaintmartin** [comment]
It seems it broke https://github.com/helm/charts/pull/10388, can you confirm this?

**Comment by desaintmartin** [comment]
It seems it broke #10388, can you confirm this?

**Pull request merged**